### ISSUE TITLE
Benchmark against the bvh crate

### ISSUE THREAD

**Issue by svenstaro** [MeirBon/rtbvh#1]
Hey, I think this is a very interesting project! I maintain the [bvh crate](https://github.com/svenstaro/bvh) and I think it'd be super interesting to see a benchmark between this. I suppose yours will be many times faster but I'm very interested to see just _how_ fast.

**Comment by meirbon** [comment]
Hey! Thanks! Nice to hear from you and I totally agree. I had actually been meaning to compare bvh libraries in Rust but just never really had the time. To be honest, my current benchmark isn’t very representative either (it just traces the same ray over and over) so I’m open to new ideas.

**Comment by meirbon** [comment]
Okay, so I spend this weekend testing a fair bit and found these things:
- Your AABB intersection code was faster than mine which I found interesting.
- Using an iterator to traverse the bvh was up to 30% faster in most cases so I implemented that in my library as well, although I do not exactly understand why (maybe compiler optimizations?). The traversal code is very similar so it must be some weird iterator optimization that the Rust compiler is able to make.
- My benchmark single ray app was not representative as I expected so I wrote a simple visualizer this morning and it gave me these numbers on a teapot:
```
bvh 0.5: ~45 MRays/s
rtbvh (spatial): ~66 MRays/s
rtbvh (spatial mbvh): ~94 MRays/s
rtbvh (binned sah): ~64 MRays/s
rtbvh (binned sah mbvh): ~91 MRays/s
```

Construction times for a teapot are as follows:
- rtbvh spatial: 27.574 ms, 0.513 ms for its mbvh
- rtbvh binned sah: 3.3 ms, 0.361 ms for its mbvh
- bvh 0.5: 10.332 ms

I have not yet added packet traversal to the visualizer, that will be my next step :)
